connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Herder.java

```diff
@@ -82,7 +82,7 @@
     void connectorConfig(String connName, Callback<Map<String, String>> callback);
 
     /**
-     * Set the configuration for a connector. This supports creation, update, and deletion.
+     * Set the configuration for a connector. This supports creation and updating.
```

Comment:
Yeah, if we're confident it's not a compatibility breaking thing, then I agree the explicit `DELETE` is better.